### DISCUSSION TITLE
Train MASK-RCNN using RLE-format mask

### DISCUSSION THREAD

**Discussion by KevinZhoutianyi** [facebookresearch/detectron2#2759] [Q&A]
## 📚 Documentation

* https://detectron2.readthedocs.io/en/latest/tutorials/datasets.html?highlight=sem_seg_file_name#standard-dataset-dicts

1. BACKGROUND: I have more than one hundred objects on one image. And there is only one class. For each image, I got one ground truth mask for all objects  
2. I follow the "train on custom dataset" part in the tutorial (https://colab.research.google.com/drive/16jcaJoc6bCFAQ96jDe2HwtXj7BMD_-m5) and I want to use bitmask instead of polygons.
3. I use pycocotools.encode  to compress my ground truth mask to RLE format. And I assign that RLE-format mask to each "segmentation" in  "annotations"
4. Finally, it is really slow when training. I guess it is because the mask is large, and there is a same mask for each object.

Is there any mistake in my steps? Or how should I use my ground truth mask image?
Thanks for your help

**Comment by ZhuangLii** [comment]
Same problem,  could you please provide any solution?
